docs/spire.md

```diff
@@ -0,0 +1,273 @@
+<!--
+---
+linkTitle: "TaskRun Result Attestation"
+weight: 1660
+---
+-->
+# TaskRun Result Attestations
```

Comment:
@jagathprakash I think you mentioned adding a caveat to this doc indicating it is a WIP, at first glance here I'm not seeing it - it might be more obvious if you remove more of this content initially? im not sure where to draw the line, i could see including some of the spire setup info and how to configure the config map that this PR is adding but i think including docs on how this would actually work when it doesn't yet work might be misleading (and would make more sense to review in the same PR that adds the functionality)

curious if @afrittoli has any thoughts on this too since we've discussed how to add docs incrementally a bit

Reply:
> But just adding a doc for a configmap seems hard. More so since it does not give any context on why we are adding the configMap.

While the implementation is ongoing, I think it would be reasonable to:
* Add documentation that links to the TEP in progress (while I don't think the final documentation should rely on the TEP, I think it's reasonable to do before the feature is complete enough to use)
* Make sure the config map itself is well documented

specifically looking at spire.md, I would suggest that this commit includes the "Architecture Overview" section and maybe "Enabling TaskRun result attestations" as well

Reply:
@bobcatfish I added the following to the documentation

-  Link to the TEP. This is added as part of the caveat that this document is a work in progress.
- More documentation on the conffig map. Specifically how the config map comes into play and the explanation of the fields in the config map.

Unchanged sections - "Architecture overview".
Retained sections - all, which includes "Architecture overview" and "Enabling TaskRun result attestations".

Reply:
hmm there's still a lot of documentation here that doesn't really have to do with the features being added in this PR - here's a gist with an example of what I had in mind: https://gist.github.com/bobcatfish/69a66c5296e3f012b7d271c0d03c3415

^^ trying to add documentation which:
* Clearly indicates the feature is not implemented
* Clearly points the reader to where the complete feature is described (would be great if that included a phased implementation approach so they could know how this will be incrementally merged but that's another story :D)
* Has documentation for what this PR includes: support for configuring Tekton Pipelines to connect to SPIRE

Reply:
Thanks for the example doc. I have updated spire.md as suggested.
Also added a list of phases for the implementation.

Reply:
that's great! thanks

Reply:
oh actually im surprised to still see so much content in the doc, in https://gist.github.com/bobcatfish/69a66c5296e3f012b7d271c0d03c3415 i was suggesting that you remove a lot of it. it seems like you feel strongly about including that here?

Reply:
@jagathprakash you can move some of the documentation out from here and make it part of the other 2 remaining PRs. Let me know if you need help to where it needs to be split up.

Reply:
Thanks @bobcatfish @pxp928 for the suggestions.
I have reduced the documentation as @bobcatfish suggested. 

Please take a look now.